tools/crm_resource.c

```diff
@@ -113,8 +119,13 @@ static gboolean
 resource_ipc_timeout(gpointer data)
 {
     // Start with newline because "Waiting for ..." message doesn't have one
-    fprintf(stderr, "\nAborting because no messages received in %d seconds\n",
-            MESSAGE_TIMEOUT_S);
+    if (error != NULL) {
+        g_clear_error(&error);
+    }
+
+    g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_TIMEOUT,
+                "Aborting because no messages received in %d seconds", MESSAGE_TIMEOUT_S);
+
     crm_err("No messages received in %d seconds", MESSAGE_TIMEOUT_S);
```

Comment:
Looking at this code again, I'm wondering why we use CMD_ERR() at all or crm_err() in this case. The log messages will only be displayed if some -V's are given, but the fprintf()'s will always go out with the same content (and both go to stderr). It was intentional in 864345a4, but I'm not sure why. That was very early in the project, so it's possible the logs went somewhere else at the time.

I'm thinking let's ditch the crm_err() here, and this PR already gets rid of the rest of the logging

Reply:
There's still a ton of uses of CMD_ERR in crm_resource_runtime.c and crm_resource_ban.c, if we wanted to take a harder look at all those.  I've removed this one use of crm_err.

Reply:
We can replace those when we eventually get to switching crm_resource to pcmk__output_t